Mathlib/RingTheory/Flat.lean

```diff
@@ -76,6 +91,121 @@ instance self (R : Type u) [CommRing R] : Flat R R :=
       lift.tmul, Submodule.subtype_apply, Algebra.id.smul_eq_mul, lsmul_apply]⟩
 #align module.flat.self Module.Flat.self
 
+variable (M : Type v) [AddCommGroup M] [Module R M]
+
+/-- A reformulation of the flat property. -/
+lemma iff_rTensor_injective :
+    Flat R M ↔ (∀ ⦃I : Ideal R⦄ (_ : I.FG), Injective (rTensor M I.subtype)) := by
+  have aux : ∀ (I : Ideal R), ((TensorProduct.lid R M).comp (rTensor M I.subtype)) =
+    (TensorProduct.lift ((lsmul R M).comp I.subtype))
+  · intro I; apply TensorProduct.ext'; intro x y; simp
+  constructor
+  · intro F I hI
+    erw [← Equiv.comp_injective _ (TensorProduct.lid R M).toEquiv]
+    have h₁ := F.out hI
+    rw [← aux] at h₁
+    refine h₁
+  · intro h₁
+    constructor
+    intro I hI
+    rw [← aux]
+    simp [h₁ hI]
+
+variable (N : Type w) [AddCommGroup N] [Module R N]
+
+/-- A retract of a flat `R`-module is flat. -/
+lemma of_retract [f : Flat R M] (i : N →ₗ[R] M) (r : M →ₗ[R] N) (h : r.comp i = LinearMap.id) :
+    Flat R N := by
+  rw [iff_rTensor_injective] at *
+  intro I hI
+  have h₁ : Function.Injective (lTensor R i)
+  · apply Function.RightInverse.injective (g := (lTensor R r))
+    intro x
+    rw [← LinearMap.comp_apply, ← lTensor_comp, h]
+    simp
+  rw [← Function.Injective.of_comp_iff h₁ (rTensor N I.subtype), ← LinearMap.coe_comp]
+  rw [LinearMap.lTensor_comp_rTensor, ←LinearMap.rTensor_comp_lTensor]
+  rw [LinearMap.coe_comp, Function.Injective.of_comp_iff (f hI)]
+  apply Function.RightInverse.injective (g := lTensor _ r)
+  intro x
+  rw [← LinearMap.comp_apply, ← lTensor_comp, h]
+  simp
+
+/-- A `R`-module isomorphic to a flat `R`-module is flat. -/
+lemma of_iso [f : Flat R M] (e : N ≃ₗ[R] M) : Flat R N := by
```

Comment:
```suggestion
lemma of_equiv [f : Flat R M] (e : N ≃ₗ[R] M) : Flat R N := by
```
We usually use this naming distinction, where `equiv` is used for (linear) equivalences, and `iso` is used for categorical isomorphisms. Of course there is no mathematical distinction, but in Lean these are two different (isomorphic) types.

Reply:
I’ll change it to `of_linearEquiv` then.